### PR TITLE
Configure the experimental testing correctly and apply minor fixes.

### PR DESCRIPTION
**Pull request by younies**
Fixes: #4596

### CODE REVIEW

File: experimental/components/Cargo.toml [hunk 1]
```diff
@@ -46,29 +46,88 @@ log = { version = "0.4", optional = true }
...
+icu_locid_transform = { workspace = true, optional = true, features = [
+    "compiled_data",
+] }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
 criterion = "0.4.0"
 
 [dev-dependencies]
 icu_locid = { path = "../../components/locid" }
+num = "0.4.1"
```
Comment (robertbastian) [inline experimental/components/Cargo.toml:60]: do not pull in the `num` crate.

File: experimental/components/tests/displaynames/tests.rs [hunk 2]
```diff
@@ -2,10 +2,12 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+extern crate alloc;
+
+use alloc::borrow::Cow;
```
Comment (robertbastian) [inline experimental/components/tests/displaynames/tests.rs:7]: Tests always have `std`
```suggestion
use std::borrow::Cow;
```

File: experimental/components/Cargo.toml (hunk 1 above)
Comment (younies) [reply experimental/components/Cargo.toml:60]: this is `[deb-dependencies]` ?

Comment (robertbastian) [reply experimental/components/Cargo.toml:60]: you're not even using it

Comment (younies) [reply experimental/components/Cargo.toml:60]: I use it in `use num_traits::sign::Signed;` because I use `abs()` fn.

Comment (robertbastian) [reply experimental/components/Cargo.toml:60]: In `use num_traits::sign::Signed;` you use `num_traits`

File: experimental/components/Cargo.toml [hunk 3]
```diff
@@ -57,7 +57,7 @@ criterion = "0.4.0"
 
 [dev-dependencies]
 icu_locid = { path = "../../components/locid" }
-num = "0.4.1"
+num-traits = "0.2.18"
```
Comment (robertbastian) [inline experimental/components/Cargo.toml:60]: you already have `num-traits` as a non-dev dependency

Comment (younies) [reply experimental/components/Cargo.toml]: deleted.

File: experimental/components/Cargo.toml [hunk 4]
```diff
@@ -45,7 +45,7 @@ displaydoc = { version = "0.2.3", default-features = false }
 log = { version = "0.4", optional = true }
 num-bigint = "0.4.4"
 num-rational = "0.4.1"
-num-traits = "0.2.17"
+num-traits = "0.2.18"
```
Comment (sffc) [inline experimental/components/Cargo.toml:48]: As discussed: please disable `default-features` on the num crates

Comment (younies) [reply experimental/components/Cargo.toml]: done

**Review by robertbastian — APPROVED** [review]

**Pull request merged**